kbfsmd/root_metadata_v3.go

```diff
@@ -645,11 +645,35 @@ func (md *RootMetadataV3) CheckValidSuccessorForServer(
 	return nil
 }
 
+// IsBackedByTeam returns true if md is for a TLF backed by a team. It could be
+// either a SingleTeam TLF or a private/public TLF backed by an implicit team.
+func (md *RootMetadataV3) IsBackedByTeam(extra ExtraMetadata) bool {
+	if extra != nil {
+		return false
+	}
+	if len(md.WriterMetadata.UnresolvedWriters) != 0 {
+		return false
+	}
+	if len(md.WriterMetadata.Writers) != 1 {
+		return false
+	}
+	if _, err := md.WriterMetadata.Writers[0].AsTeam(); err != nil {
+		return false
+	}
+	return true
+}
+
 // MakeBareTlfHandle implements the RootMetadata interface for RootMetadataV3.
 func (md *RootMetadataV3) MakeBareTlfHandle(extra ExtraMetadata) (
 	tlf.Handle, error) {
+	if md.IsBackedByTeam(extra) {
+		return tlf.MakeHandle(md.WriterMetadata.Writers,
+			nil, nil, nil, md.TlfHandleExtensions())
+	}
+
 	var writers, readers []keybase1.UserOrTeamID
-	if md.TlfID().Type() == tlf.Private {
+	switch md.TlfID().Type() {
```

Comment:
My idea was to add a method like this following to the `kbfsmd.RootMetadata` interface:

```go
TypeForKeying() tlf.Type
```

The v3 implementation would be something like:

```go
func (md *RootMetadataV3) TypeForKeying() tlf.Type {
  if md.IsBackedByTeam() {  // Would need to drop the `extra` argument
    return tlf.SingleTeam
  }
  return tlf.Type()
}
```

Then we can just replace `md.TlfID().Type()` with `md.TypeForKeying()` everywhere that an implicit team TLF should be treated as a team TLF.  This would make KBFS-2600 much easier.  What do you think?

Reply:
Oh I see; I guess that makes sense, although it's a bit confusing to use `tlf.SingleTeam`. Would it be similar amount of work to have these in `tlf` and use them instead, when what we care is TypeForKeying rather than the actual TLF type?

```
type TypeForKeying Type
const (
    UnknownKeyingType TypeForKeying = iota
    PrivateKeying
    PublicKeying
    TeamKeying
)
```

If not, then I'm OK with your proposed approach too as long as there are comments to remind us of the design!

Reply:
Sure, I guess adding that new type isn't too much more work.  Sounds good, thanks!

Reply:
Cool; will do. thank